.github/workflows/test.yml

```diff
@@ -75,97 +76,79 @@ jobs:
         # https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations
         include:
           # test default settings with 3.11 across all supported evm versions
-          - python-version: ["3.11", "311"]
-            debug: false
+          - debug: false
             opt-mode: gas
             evm-version: london
-            evm-backend: revm
 
-          - python-version: ["3.11", "311"]
-            debug: false
+          - debug: false
             opt-mode: gas
             evm-version: paris
-            evm-backend: revm
-
-          # redundant rule, for clarity
-          - python-version: ["3.11", "311"]
-            debug: false
-            opt-mode: gas
-            evm-version: shanghai
-            evm-backend: revm
 
-          - python-version: ["3.11", "311"]
-            debug: false
+          - debug: false
             opt-mode: gas
             evm-version: cancun
-            evm-backend: revm
 
           # py-evm rules
-          - python-version: ["3.11", "311"]
-            debug: false
+          - debug: false
             opt-mode: codesize
             evm-version: london
             evm-backend: py-evm
 
-          - python-version: ["3.11", "311"]
-            debug: false
+          - debug: false
             opt-mode: gas
             evm-version: cancun
             evm-backend: py-evm
 
           # test experimental pipeline
-          - python-version: ["3.11", "311"]
-            opt-mode: gas
+          - opt-mode: gas
             debug: false
-            evm-version: shanghai
-            evm-backend: revm
             experimental-codegen: true
 
-          - python-version: ["3.11", "311"]
-            opt-mode: codesize
+          - opt-mode: codesize
             debug: false
-            evm-version: shanghai
-            evm-backend: revm
             experimental-codegen: true
 
-          - python-version: ["3.11", "311"]
-            opt-mode: none
+          - opt-mode: none
             debug: false
-            evm-version: shanghai
-            evm-backend: revm
             experimental-codegen: true
 
+          # os-specific rules
+          - os: windows
+            opt-mode: none
```

Comment:
i don't think we should override the default -- as much as possible, we should try to vary only on a single axis. (like we want to test the most common use case, which will be running with default optimization level)